packages/core/src/app/customer/EmailField.tsx

```diff
@@ -1,36 +1,42 @@
 import { FieldProps } from 'formik';
 import React, { FunctionComponent, memo, useCallback, useMemo } from 'react';
 
+import { isFloatingLabelEnabled } from "../common/utility";
+import useCheckoutContext from "../common/utility/useCheckoutContext";
 import { TranslatedString } from '../locale';
 import { FormField, TextInput } from '../ui/form';
+import Label from "../ui/form/Label";
 
 export interface EmailFieldProps {
     onChange?(value: string): void;
 }
 
-const EmailField: FunctionComponent<EmailFieldProps> = ({ onChange }) => {
-    const renderInput = useCallback(
-        (props: FieldProps) => (
-            <TextInput
-                {...props.field}
-                autoComplete={props.field.name}
-                id={props.field.name}
-                type="email"
-            />
-        ),
-        [],
-    );
+const EmailField: FunctionComponent<EmailFieldProps>  = ({
+    onChange,
+}) => {
+    const renderInput = useCallback((props: FieldProps) => (
+        <TextInput
+            { ...props.field }
+            autoComplete={ props.field.name }
+            id={ props.field.name }
+            type="email"
+        />
+    ), []);
 
-    const labelContent = useMemo(() => <TranslatedString id="customer.email_label" />, []);
+    const useFloatingLabel = isFloatingLabelEnabled(useCheckoutContext());
```

Comment:
I used `isFloatingLabelEnabled()` helper function to switch floating lables on/off.